File: Design-Patterns/Bridge/CSharp/Bridge/Spells/FireSpell.cs
using System;

namespace Bridge.Spells
{
    public class FireSpell : Spell
    {
        public FireSpell(int power, int speed, int cost) : base(power, speed, cost)
        {

        }

        public override void Cast()
        {
            Console.WriteLine($"Casts a fire spell with power {Power} and speed {Speed}. Applies burning effect.");
        }
    }
}

File: Design-Patterns/Bridge/Cpp/fire_spell.cpp
#include <iostream>
#include <string>
#include "spell.h"

fire_spell::fire_spell(int power, int speed, int cost) : spell(power, speed, cost)
{

}

void fire_spell::cast()
{
	std::cout << "Casts a fire spell with power " + std::to_string(power) + " and speed " + std::to_string(speed) + ". Applies burning effect." << std::endl;
}